Automattic/jetpack
# Sharing Plugin: Keep polishing implementation

This is a follow-up PR of https://github.com/Automattic/jetpack/pull/20944.

Enhancements

* Replace HOC with hooks
* Pick up store definition from a package instead of hardcoding its definition.
* Move components into its proper folder

Related https://github.com/Automattic/jetpack/issues/16816

#### Changes proposed in this Pull Request:

* Polish plugin sharing implementation

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Activate the `Like buttons` module from https://<your-deb-site>/wp-admin/admin.php?page=jetpack#/sharing
![image](https://user-images.githubusercontent.com/77539/132554969-35a00de2-a7d9-426e-91b0-32535440e516.png)

* Create a new post
* Confirm it's rendered in the Jetpack sidebar rightly.
<img src="https://user-images.githubusercontent.com/77539/132555359-6d29dac6-aa3e-4a83-a2f8-bb55257f4522.png" width="400px" />
* Confirm the `Show sharing buttons` value is properly updated by clicking on the proper checkbox after saving the post. Additionally, You can take a look at the async request from the dev tool - network tab:

<img width="600" alt="Screen Shot 2021-09-08 at 2 25 09 PM" src="https://user-images.githubusercontent.com/77539/132556254-ac655310-4565-41fb-aa95-0c3463351fd2.png">


* In the end, simply confirm the feature keeps working as expected :smart:
